src/gcp/cloudfunctionsv2.ts

```diff
@@ -386,6 +394,19 @@ export function functionFromSpec(cloudFunction: backend.FunctionSpec, source: St
     };
     if (gcfFunction.eventTrigger.eventType === PUBSUB_PUBLISH_EVENT) {
       gcfFunction.eventTrigger.pubsubTopic = cloudFunction.trigger.eventFilters.resource;
+    } else if (GCS_EVENTS.has(gcfFunction.eventTrigger.eventType)) {
```

Comment:
I see why you extended `else if` off of the previous `if`, but Pub/Sub _must_ be represented differently in GCFv2, GCS doesn't. If our hack is here, then the container contract _also_ needs to use `resource` instead of `bucket`, which makes this everlasting techncial debt.

Instead, please put logic in triggerParser to detect the event is a GCS event and store the TriggerAnnotation's `resource` as the `FunctionSpec` and `Endpoint`'s `bucket`. That way the GCP folder can stay simpler.

Reply:
Moved all this logic into triggerParser